### DISCUSSION TITLE
Firebase analytics auto collection behavior

### DISCUSSION THREAD

**Discussion by toytonic** [invertase/react-native-firebase#5278] [Q&A]
Hi there,
I have some questions regarding analytics auto collection behavior. Maybe someone knows and can provide a little help/guidance here.

We disabled analytics auto collection by default both via `firebase.json` and the respective native settings in `Info.plist` and `AndroidManifest.xml`. The user can later on agree to enable tracking during registration.

a) However we still see `first_open` events in our stats. How is that possible? Does firebase remember these events and logs once collection is enabled via Javascript?

b) I tried digging with the Debug Build Flag for Firebase enabled. It logs all sorts of events regardless of having auto collection disabled by default. Is the flag not taken into account in debug mode?

Thanks a lot for your help

**Comment by mikehardy** [comment]
Hi there - I am *guessing* the events are queued - but we do not do any special processing in react-native-firebase between debug and release. You mention "Debug Build Flag", that's ambiguous for me. Do you mean you turned on analytics debugging via the `adb setprop` command on Android or with a flag (please specify if so) on iOS? And when you say "it logs" do you mean you saw it in the logs or do you mean you actually see it on google analytics in your app's analytics console?

Obviously if you have collection disabled, you should not see it on the app analytics console on the web. Any other place you see it is inconsequential as it hasn't left the device. If you then enable it and see first_opens but only after enable, then that's data that would fit a queuing hypothesis.

**Reply by toytonic** [reply]
Hi @mikehardy, thanks a lot for your answer.
Some clarifications: With Debug Flag I mean the `-FIRDebugEnabled` Flag for iOS set via xcode. With logs I mean, events popping up in the debug view in the firebase console.

From my tests it looks a lot like the firebase sdk is queuing events. On the other hand when I saw my simulator 'connected' in the debug view within the firebase console. It looked like the setting `analytics_auto_collection_enabled: false` did not have any effect here as I was still seeing auto logged events there.

**Reply by mikehardy** [reply]
Hmm - can you log a real issue this so it doesn't get lost? discussions are good for batting ideas around but I can't label them for or workflow queues. When you do so you can link here but please post at minimum the versions you are on of react-native-firebase, the exact way you tested (firebase.json setting, any API calls, what flag you launched with on iOS, and what you see in the Xcode console logs + the firebase analytics web console). Then we can see if we reproduce it. It won't get lost then but it may still be a little bit - the underlying SDKs have thrown an enormous amount of change at us in a short amount of time and there's a lot of high priority work, so any continued investigation you do (e.g., on android to check it, or double-checking iOS by looking at the actual console logs) would help advance it in the meantime
